presto-main/src/main/java/com/facebook/presto/operator/scalar/ReflectionParametricScalar.java

```diff
@@ -173,40 +181,82 @@ public ScalarFunctionImplementation specialize(BoundVariables boundVariables, in
         throw new PrestoException(FUNCTION_IMPLEMENTATION_MISSING, format("Unsupported type parameters (%s) for %s", boundVariables, getSignature()));
     }
 
-    public static SqlScalarFunction parseDefinition(Class<?> clazz)
+    public static List<SqlScalarFunction> parseDefinition(Class<?> clazz)
     {
         ScalarFunction scalarAnnotation = clazz.getAnnotation(ScalarFunction.class);
         ScalarOperator operatorAnnotation = clazz.getAnnotation(ScalarOperator.class);
         Description descriptionAnnotation = clazz.getAnnotation(Description.class);
-        checkArgument(scalarAnnotation != null || operatorAnnotation != null, "Missing parametric annotation");
-        checkArgument(scalarAnnotation == null || operatorAnnotation == null, "%s annotated as both an operator and a function", clazz.getSimpleName());
+
         if (scalarAnnotation != null) {
-            requireNonNull(descriptionAnnotation, format("%s missing @Description annotation", clazz.getSimpleName()));
+            return ImmutableList.of(createReflectionParametricScalar(scalarAnnotation.value(), scalarAnnotation.hidden(), scalarAnnotation.deterministic(), descriptionAnnotation, findPublicMethodsWithAnnotation(clazz, SqlType.class), clazz.getSimpleName(), findConstructors(clazz)));
         }
-        String name;
-        String description = descriptionAnnotation == null ? "" : descriptionAnnotation.value();
-        boolean hidden;
-        boolean deterministic;
+        else if (operatorAnnotation != null) {
+            return ImmutableList.of(createReflectionParametricScalar(mangleOperatorName(operatorAnnotation.value()), true, true, descriptionAnnotation, findPublicMethodsWithAnnotation(clazz, SqlType.class), clazz.getSimpleName(), findConstructors(clazz)));
+        }
+        checkArgument(false);
+        return ImmutableList.of();
```

Comment:
This is not reachable. Instead change the `checkArgument` to `throw new ...` with an appropriate message